### ISSUE TITLE
LoRA finetuning

### ISSUE THREAD

**Issue by xiamengzhou** [allenai/open-instruct#5]
Hi,

Thanks for releasing this amazing repo! In the repo, a script to finetune with LoRA has been provided, yet results with LoRA finetuning is not mentioned in the paper. Is LoRA finetuning underperforming full finetuning?

Best,
Mengzhou

**Comment by hamishivi** [comment]
Hi,

We experimented a bit with LoRA finetuning initially, but sort of left it aside after a bit since we had the compute budget to fully-finetune. We don't have comprehensive results, but it seemed LoRA training generally did about as well as full finetuning with a large enough rank (from memory, we tried with something like 128).
Hope that's useful!

**Comment by kelvennn** [comment]
> Hi,
> 
> We experimented a bit with LoRA finetuning initially, but sort of left it aside after a bit since we had the compute budget to fully-finetune. We don't have comprehensive results, but it seemed LoRA training generally did about as well as full finetuning with a large enough rank (from memory, we tried with something like 128). Hope that's useful!

Hi,
Do you have the effecient setting of hyperparameters of LoRA been recorded?  Thank you.

**Comment by hamishivi** [comment]
Hi,

I spent the past week or two adding LoRA + QLoRA into the repo, and found that using a rank of 256, alpha=256, learning rate=1e-4, and training for 5 epochs works well, and results in a model (on 7/13b scales) that performs similarly to full-finetuning except in AlpacaFarm evaluation, where the fully-finetuned model does slightly better.

Check out scripts/finetune_lora_with_accelerate.sh and scripts/finetune_qlora_with_accelerate.sh for examples on how to use the PEFT tuning, and our recommended hyperparameters.

Thanks!
